spark/dl/src/main/scala/com/intel/analytics/bigdl/dataset/MiniBatch.scala

```diff
@@ -95,11 +137,299 @@ class TensorMiniBatch[T](
+    var s = 0
+    while (s < samples.length) {
+      var f = 0
+      var offset = 0
+      val featureSize = featureSizes(s)
+      while (f < inputs.length) {
+        val length = featureSize(f).product
+        if (featurePadding.isDefined) {
+          // copy data
+          copy(samples(s).getData(), offset,
+            length, inputs(f)(s + 1), featurePadding.get(f))
+        } else {
+          // copy data without padding.
+          copy(samples(s).getData(), offset,
+            length, inputs(f)(s + 1))
+        }
+        f += 1
+        offset += length
+      }
+
+      if (!unlabeled) {
+        var l = 0
+        val labelSize = labelSizes(s)
+        while (l < target.length) {
+          val length = labelSize(l).product
+          copy(samples(s).getData(), offset,
+            length, target(l)(s + 1))
+          l += 1
+          offset += length
+        }
+      }
+
+      s += 1
+    }
+
+    miniBatch
+  }
+
+  /**
+   * Find Sample in Array[Sample] who has the biggest featureLength
+   */
+  private[bigdl] def findLongestFeatures[T: ClassTag](
+        samples: Array[Sample[T]])(implicit ev: TensorNumeric[T]): Array[Int] = {
+    val featureIndices =
+      new Array[Int](samples(0).featureLength().length).map(_ => 0)
+    var i = 1
+    while (i < samples.length) {
+      var j = 0
+      while (j < featureIndices.length) {
+        if (samples(i).featureLength()(j) > samples(featureIndices(j)).featureLength()(j)) {
+          featureIndices(j) = i
+        }
+        j += 1
+      }
+      i += 1
+    }
+    featureIndices
+  }
+
+  /**
+   * Find Sample in Array[Sample] who has the biggest labelLength
+   */
+  private[bigdl] def findLongestLabels[T: ClassTag](
+        samples: Array[Sample[T]])(implicit ev: TensorNumeric[T]): Array[Int] = {
+    val labelIndices =
+      new Array[Int](samples(0).labelLength().length).map(_ => 0)
+    var i = 1
+    while (i < samples.length) {
+      var j = 0
+      while (j < labelIndices.length) {
+        if (samples(i).labelLength()(j) > samples(labelIndices(j)).labelLength()(j)) {
+          labelIndices(j) = i
+        }
+        j += 1
+      }
+      i += 1
+    }
+    labelIndices
+  }
+
+  /**
+   * Copy tensor src to tensor dest with a padding tensor.
+   */
+  private def copy[T: ClassTag](
+      src: Array[T],
+      offset: Int,
+      length: Int,
+      dest: Tensor[T],
+      paddingTensor: Tensor[T] = null)(implicit ev: TensorNumeric[T]): Unit = {
+    ev.arraycopy(src,
+      offset,
+      dest.storage().array(),
+      dest.storageOffset() - 1,
+      length)
+    if (null != paddingTensor) {
+      var j = length
+      while (j < dest.nElement()) {
+        ev.arraycopy(paddingTensor.storage().array(), paddingTensor.storageOffset() - 1,
+          dest.storage().array(), dest.storageOffset() - 1 + j, paddingTensor.nElement())
+        j += paddingTensor.nElement()
+      }
+    }
   }
 }
 
+/**
+ * Padding param for ArrayTensorMiniBatch.
```

Comment:
you should tell user in the doc that it will append tensor to the first dimension